From a1e3ec14bc7bf5a127d99ce3c2dc2558531a06e2 Mon Sep 17 00:00:00 2001
From: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Date: Tue, 25 Sep 2018 23:23:31 +0530
Subject: [PATCH 0121/1852] ASoC: xlnx: correct the audio formatter masks

commit 1c8939fa051a83ac5288bacc2a18afcdba2cd232 from
https://github.com/Xilinx/linux-xlnx.git

Corrected few mask and shift values of audio formatter.

Signed-off-by: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_formatter_pcm.c | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/sound/soc/xilinx/xlnx_formatter_pcm.c b/sound/soc/xilinx/xlnx_formatter_pcm.c
index 0b6471b41b16..dc3351e2d0de 100644
--- a/sound/soc/xilinx/xlnx_formatter_pcm.c
+++ b/sound/soc/xilinx/xlnx_formatter_pcm.c
@@ -45,15 +45,15 @@
 
 #define CFG_MM2S_CH_MASK	GENMASK(11, 8)
 #define CFG_MM2S_CH_SHIFT	8
-#define CFG_MM2S_FORMAT_MASK	GENMASK(14, 13)
-#define CGG_MM2S_FORMAT_SHIFT	13
-#define CGG_MM2S_PKG_MASK	BIT(12)
+#define CFG_MM2S_XFER_MASK	GENMASK(14, 13)
+#define CFG_MM2S_XFER_SHIFT	13
+#define CFG_MM2S_PKG_MASK	BIT(12)
 
 #define CFG_S2MM_CH_MASK	GENMASK(27, 24)
 #define CFG_S2MM_CH_SHIFT	24
-#define CFG_S2MM_FORMAT_MASK	GENMASK(30, 29)
-#define CGG_S2MM_FORMAT_SHIFT	29
-#define CGG_S2MM_PKG_MASK	BIT(28)
+#define CFG_S2MM_XFER_MASK	GENMASK(30, 29)
+#define CFG_S2MM_XFER_SHIFT	29
+#define CFG_S2MM_PKG_MASK	BIT(28)
 
 #define AUD_CTRL_DATA_WIDTH_SHIFT	16
 #define AUD_CTRL_ACTIVE_CH_SHIFT	19
@@ -189,18 +189,18 @@ static int xlnx_formatter_pcm_open(struct snd_pcm_substream *substream)
 	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
 		ch_count_mask = CFG_MM2S_CH_MASK;
 		ch_count_shift = CFG_MM2S_CH_SHIFT;
-		data_format_mode = CFG_MM2S_FORMAT_MASK;
-		data_xfer_mode = CGG_MM2S_FORMAT_SHIFT;
-		data_xfer_shift = CGG_MM2S_PKG_MASK;
+		data_xfer_mode = CFG_MM2S_XFER_MASK;
+		data_xfer_shift = CFG_MM2S_XFER_SHIFT;
+		data_format_mode = CFG_MM2S_PKG_MASK;
 		stream_data->mmio = adata->mmio + XLNX_MM2S_OFFSET;
 		adata->play_stream = substream;
 
 	} else {
 		ch_count_mask = CFG_S2MM_CH_MASK;
 		ch_count_shift = CFG_S2MM_CH_SHIFT;
-		data_format_mode = CFG_S2MM_FORMAT_MASK;
-		data_xfer_mode = CGG_S2MM_FORMAT_SHIFT;
-		data_xfer_shift = CGG_S2MM_PKG_MASK;
+		data_xfer_mode = CFG_S2MM_XFER_MASK;
+		data_xfer_shift = CFG_S2MM_XFER_SHIFT;
+		data_format_mode = CFG_S2MM_PKG_MASK;
 		stream_data->mmio = adata->mmio + XLNX_S2MM_OFFSET;
 		adata->capture_stream = substream;
 	}
-- 
2.31.1

